staging/src/k8s.io/apiserver/pkg/admission/plugin/namespace/lifecycle/admission.go

```diff
@@ -140,7 +140,7 @@ func (l *Lifecycle) Admit(ctx context.Context, a admission.Attributes, o admissi
 			exists = true
 		}
 		if exists {
-			klog.V(4).Infof("found %s in cache after waiting", a.GetNamespace())
+			klog.V(4).InfoS("namespace exists in cache after waiting", "ns", a.GetNamespace())
```

Comment:
```suggestion
			klog.V(4).InfoS("namespace exists in cache after waiting", "namespace", a.GetNamespace())
```

I think we should prefer uniformity of using full resource type names than benefit of saving on writing couple of bytes.

Reply:
I think that namespace should be treated here as a Resource Type instead of just name. As a resource type we should use `klog.KRef` to make it proper reference to Resource. Empty string `""` is passed as namespace is non-namespaced resource type.

Reply:
I haved updated the code, but the  a.GetNamespace() return namespace here, so updated to klog.KRef(a.GetNamespace(), "") here

Reply:
I still think it should be `klog.KRef("", a.GetNamespace())`. Most object reference is build from pair Namespace + Name, there are also non-namespaced objects like Node which only have name. 

Namespace in K8s is not only a way to refer objects, but also it is a normal K8s object too. In this case namespace is an object with only a name (non-namespaced one like Node). When printing information about Namespace we should pass it's name into "name" field and leave "namespace" field empty, and not reverse as in your case.